File: ITapply/ITapply.Services/Database/CVDocument.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ITapply.Services.Database
{
    public class CVDocument
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int CandidateId { get; set; }
        [ForeignKey("CandidateId")]
        public Candidate Candidate { get; set; }

        [Required]
        [StringLength(255)]
        public string FileName { get; set; }

        [Required]
        // Max file size is configured in DbContext (5MB)
        public byte[] FileContent { get; set; }

        public bool IsMain { get; set; } = false;

        [Required]
        public DateTime UploadDate { get; set; } = DateTime.Now;
    }
}

File: ITapply/ITapply.UI/itapply_mobile/lib/models/cv_document.dart
import 'package:json_annotation/json_annotation.dart';

part 'cv_document.g.dart';
@JsonSerializable()
class CVDocument {
  final int id;
  final int candidateId;
  final String? candidateName;
  final String fileName;
  final String fileContent;
  final bool isMain;
  final DateTime uploadDate;

  CVDocument({
    required this.id,
    required this.candidateId,
    this.candidateName,
    required this.fileName,
    required this.fileContent,
    required this.isMain,
    required this.uploadDate,
  });

  factory CVDocument.fromJson(Map<String, dynamic> json) => _$CVDocumentFromJson(json);
  Map<String, dynamic> toJson() => _$CVDocumentToJson(this);
}